### PR TITLE
add markDynamoStrictTest to TestOptimRenewed, removing flakiness

### LINKED ISSUE #115391: DISABLED test_foreach_matches_forloop_RMSprop_cpu_float64 (__main__.TestOptimRenewedCPU)
Platforms: dynamo

This test was disabled because it is failing in CI. See [recent examples](https://hud.pytorch.org/flakytest?name=test_foreach_matches_forloop_RMSprop_cpu_float64&suite=TestOptimRenewedCPU) and the most recent trunk [workflow logs](https://github.com/pytorch/pytorch/runs/19425649632).

Over the past 3 hours, it has been determined flaky in 12 workflow(s) with 12 failures and 12 successes.

**Debugging instructions (after clicking on the recent samples link):**
DO NOT ASSUME THINGS ARE OKAY IF THE CI IS GREEN. We now shield flaky tests from developers so CI will thus be green but it will be harder to parse the logs.
To find relevant log snippets:
1. Click on the workflow logs linked above
2. Click on the Test step of the job so that it is expanded. Otherwise, the grepping will not work.
3. Grep for `test_foreach_matches_forloop_RMSprop_cpu_float64`
4. There should be several instances run (as flaky tests are rerun in CI) from which you can study the logs.

Test file path: `test_optim.py`

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### LINKED ISSUE #115392: DISABLED test_foreach_matches_forloop_Rprop_cpu_float64 (__main__.TestOptimRenewedCPU)
Platforms: dynamo

This test was disabled because it is failing in CI. See [recent examples](https://hud.pytorch.org/flakytest?name=test_foreach_matches_forloop_Rprop_cpu_float64&suite=TestOptimRenewedCPU) and the most recent trunk [workflow logs](https://github.com/pytorch/pytorch/runs/19425649632).

Over the past 3 hours, it has been determined flaky in 12 workflow(s) with 12 failures and 12 successes.

**Debugging instructions (after clicking on the recent samples link):**
DO NOT ASSUME THINGS ARE OKAY IF THE CI IS GREEN. We now shield flaky tests from developers so CI will thus be green but it will be harder to parse the logs.
To find relevant log snippets:
1. Click on the workflow logs linked above
2. Click on the Test step of the job so that it is expanded. Otherwise, the grepping will not work.
3. Grep for `test_foreach_matches_forloop_Rprop_cpu_float64`
4. There should be several instances run (as flaky tests are rerun in CI) from which you can study the logs.

Test file path: `test_optim.py`

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### LINKED ISSUE #115393: DISABLED test_foreach_matches_forloop_Adadelta_cpu_float64 (__main__.TestOptimRenewedCPU)
Platforms: dynamo

This test was disabled because it is failing in CI. See [recent examples](https://hud.pytorch.org/flakytest?name=test_foreach_matches_forloop_Adadelta_cpu_float64&suite=TestOptimRenewedCPU) and the most recent trunk [workflow logs](https://github.com/pytorch/pytorch/runs/19425649632).

Over the past 3 hours, it has been determined flaky in 12 workflow(s) with 12 failures and 12 successes.

**Debugging instructions (after clicking on the recent samples link):**
DO NOT ASSUME THINGS ARE OKAY IF THE CI IS GREEN. We now shield flaky tests from developers so CI will thus be green but it will be harder to parse the logs.
To find relevant log snippets:
1. Click on the workflow logs linked above
2. Click on the Test step of the job so that it is expanded. Otherwise, the grepping will not work.
3. Grep for `test_foreach_matches_forloop_Adadelta_cpu_float64`
4. There should be several instances run (as flaky tests are rerun in CI) from which you can study the logs.

Test file path: `test_optim.py`

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### LINKED ISSUE #115394: DISABLED test_foreach_matches_forloop_ASGD_cpu_float64 (__main__.TestOptimRenewedCPU)
Platforms: dynamo

This test was disabled because it is failing in CI. See [recent examples](https://hud.pytorch.org/flakytest?name=test_foreach_matches_forloop_ASGD_cpu_float64&suite=TestOptimRenewedCPU) and the most recent trunk [workflow logs](https://github.com/pytorch/pytorch/runs/19425649632).

Over the past 3 hours, it has been determined flaky in 12 workflow(s) with 12 failures and 12 successes.

**Debugging instructions (after clicking on the recent samples link):**
DO NOT ASSUME THINGS ARE OKAY IF THE CI IS GREEN. We now shield flaky tests from developers so CI will thus be green but it will be harder to parse the logs.
To find relevant log snippets:
1. Click on the workflow logs linked above
2. Click on the Test step of the job so that it is expanded. Otherwise, the grepping will not work.
3. Grep for `test_foreach_matches_forloop_ASGD_cpu_float64`
4. There should be several instances run (as flaky tests are rerun in CI) from which you can study the logs.

Test file path: `test_optim.py`

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### LINKED ISSUE #115406: DISABLED test_foreach_matches_forloop_Adam_cpu_float64 (__main__.TestOptimRenewedCPU)
Platforms: dynamo

This test was disabled because it is failing in CI. See [recent examples](https://hud.pytorch.org/flakytest?name=test_foreach_matches_forloop_Adam_cpu_float64&suite=TestOptimRenewedCPU) and the most recent trunk [workflow logs](https://github.com/pytorch/pytorch/runs/19435572444).

Over the past 3 hours, it has been determined flaky in 6 workflow(s) with 6 failures and 6 successes.

**Debugging instructions (after clicking on the recent samples link):**
DO NOT ASSUME THINGS ARE OKAY IF THE CI IS GREEN. We now shield flaky tests from developers so CI will thus be green but it will be harder to parse the logs.
To find relevant log snippets:
1. Click on the workflow logs linked above
2. Click on the Test step of the job so that it is expanded. Otherwise, the grepping will not work.
3. Grep for `test_foreach_matches_forloop_Adam_cpu_float64`
4. There should be several instances run (as flaky tests are rerun in CI) from which you can study the logs.

Test file path: `test_optim.py`

cc @vincentqb @jbschlosser @albanD @janeyx99 @crcrpar @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

### PR DESCRIPTION
**Pull request by janeyx99**
fixes #115406 fixes #115394 fixes #115393 fixes #115392 fixes #115391

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115947

### CODE REVIEW

File: torch/testing/_internal/common_optimizers.py [hunk 1]
```diff
@@ -857,12 +892,28 @@ def optim_error_inputs_func_sparseadam(device, dtype):
         optim_inputs_func=optim_inputs_func_rmsprop,
         optim_error_inputs_func=optim_error_inputs_func_rmsprop,
         supported_impls=("foreach", "differentiable"),
+        skips=(
+            # See https://github.com/pytorch/pytorch/issues/115679
```
Comment (albanD) [inline torch/testing/_internal/common_optimizers.py:896]: nit: this one and the one below: move the issue into the skip message like you do above.

**Review by albanD — APPROVED** [review]
SGTM

File: torch/testing/_internal/common_optimizers.py (hunk 1 above)
Comment (janeyx99) [reply torch/testing/_internal/common_optimizers.py:896]: oh wow i forgot i was in the middle of doing that

**Review by zou3519 — APPROVED** [review]

**Comment by janeyx99** [comment]
@pytorchbot merge -r

failures are due to prev reverted pr

**Pull request closed without merging**